sdk/search/search/src/indexDocumentsBatch.ts

```diff
@@ -0,0 +1,78 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { IndexAction } from "./indexModels";
+
+export class IndexDocumentsBatch<T> {
+  /**
+   * The set of actions taken in this batch.
+   */
+  public readonly actions: IndexAction<T>[];
+
+  constructor(actions: IndexAction<T>[] = []) {
```

Comment:
In .NET, @tg-msft hide `IndexAction` and defined it as `IndexDocumentAction` for similar naming. See https://github.com/Azure/azure-sdk-for-net/blob/cc3c60778e61403dc1d7f6e4168474fcbdf7bec3/sdk/search/Azure.Search.Documents/src/Models/IndexDocumentsBatch%7BT%7D.cs#L80. It's similar in nature to this, but defines static helpers that return an `IndexDocumentBatch`. That said, we can unify on APIs across languages later.

Reply:
I'm fine renaming `IndexAction`; this isn't a type we expect TS users to directly reference so the name isn't very important to us.